pallas-crypto/src/vrf/mod.rs

```diff
@@ -0,0 +1,74 @@
+use thiserror::Error;
+use vrf_dalek::vrf03::{PublicKey03, SecretKey03, VrfProof03};
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("TryFromSlice {0}")]
+    TryFromSlice(#[from] std::array::TryFromSliceError),
+
+    #[error("VrfError {0}")]
+    VrfError(#[from] vrf_dalek::errors::VrfError),
+}
+
+/// Sign a seed value with a vrf secret key and produce a proof signature
+pub fn vrf_prove(secret_key: &[u8], seed: &[u8]) -> Result<Vec<u8>, Error> {
+    let sk = SecretKey03::from_bytes(secret_key[..32].try_into()?);
```

Comment:
disclaimer: _I have nothing against free functions_ (™️ )

why not adding new types for the `SecretKey` and `VerificationKey` ? That way you can wrap up the vrf_dalek dependency and eventually allow us to switch things around without touching the APIs too much.

also would allow us to add some more conversion API (like the `try_into` but also a memory scrub on drop).

Reply:
I've added the wrappers VrfSecretKey and VrfPublicKey. Also added the scrub on drop for the SecretKey and some From traits.